### ISSUE TITLE
Moving to gitlab

### ISSUE THREAD

**Issue by SirCmpwn** [swaywm/sway#22]
Strongly considering moving this project to Gitlab. Leave your opinions here. Advantages:
- Gitlab is open source
- Gitlab has free, well integrated CI including support for self-hosted workers
- I prefer Gitlab's politics

**Comment by Luminarys** [comment]
I agree with all of these things, but you do have to consider the primary disadvantage of Gitlab having a much smaller user base. The good, and perhaps also bad, thing about Github is its ubiquity, which results in many users being able to find your project and contribute to it. But if you're ok with the decreased visibility, then I'd say go for it.

**Comment by SirCmpwn** [comment]
Needs more discussion, folks. Speak up.

**Comment by onny** [comment]
Lets stay on Github until atleast one stable release is made. After that, the project might be "mature" enough to move on GitLab ... but now, we'll need as much contribution as possible

**Comment by SirCmpwn** [comment]
I'm not sure we do need a ton of contributions. We've already got enough consistent contributors to build an i3 clone in a fairly short amount of time.

**Comment by SirCmpwn** [comment]
I've found that most of the time, even projects with lots of contributors and exposure will be written almost entirely by one or two people.

**Comment by StianTheDark** [comment]
Gitlab sounds great, but I'm a little confused; is Gitlab self hosted or only on the web?

**Comment by SirCmpwn** [comment]
It's both. There's a public gitlab running on gitlab.com, but the code is open source and you can deploy it on your own infrastructure if you want. We'd be using gitlab.com.

**Comment by StianTheDark** [comment]
I'd say go for it :+1:

**Comment by StianTheDark** [comment]
Because its an alternative to Github, I guess they've added migration options.

**Comment by SirCmpwn** [comment]
Eh, for now let's just stay here.

**Comment by RyanDwyer** [comment]
Yeah, it's time to reconsider this. I support a move to GitLab, even if Microsoft isn't buying them out. But I'd probably wait a while to make sure GitLab has sorted out any infrastructure issues that arise.

**Comment by Hyleus** [comment]
With the recent news about Microsoft acquiring Github, I believe it's time that Sway should consider to move somewhere else, maybe Gitlab could be considered? The software is itself FOSS. I believe that projects like Sway, which are free software, should be hosted itself on a platform which is free and isn't run by a mega corporation, which was/is in the past very harmful to software in general.

**Comment by Ongy** [comment]
I'll give my 2ct here, mainly as `wlroots` consumer, and small time contributer, since I highly expect wlroots to move with sway if it does move.

I'm personally indifferent to github. I have been until now and will be for now. I think it's one of the more fleshed out platforms, but I dislike how it centralises everything at a single company.
I personally don't expect a microsoft takeover to change anything in the near future.


Afaik the long term infrastructure for sway and related projects is sr.ht, which makes the current situation a bit more complicated.
IMO sr.ht isn't ready for a big project yet. It can host, but there's missing infrastructure for public review, and public tracking of current ongoing work.
For this reason I'm not currently in favor of moving there.

On the other hand, moving to gitlab at this point will probably require another move during the next couple of years, once sr.ht is at the point where it's suitable for sway.
Since moving projects leaves dangling pointers (i.e. old hpyerlinks) and is generally an annoying effort (migrating accounts, issues) I don't think it's a good idea to move now and then again later without a pressing need.

I propose to continue as before, long term plan to migrate but don't do anything rash.
Should there be any change to the platform after a buyout that reinforces reasons to leave, I'll gladly help out to do so.

**Comment by Hyleus** [comment]
>Since moving projects leaves dangling pointers (i.e. old hpyerlinks) and is generally an annoying effort (migrating accounts, issues) I don't think it's a good idea to move now and then again later without a pressing need.

There's an Gitlab import tool, which simplifies the process, it should be able to import issues, PRs and everything else automatically. There's a video which demonstrates it: https://www.youtube.com/watch?v=VYOXuOg9tQI

**Comment by emersion** [comment]
As a contributor who probably wouldn't have started contributing if the project was using `git-send-email` on an obscure mailing list, I'd like to highlight that we need contributors and that lowering the barrier to entry is important. GitHub does a good job at this - I haven't really been using GitLab but it seems good enough.

Yeah, the plan is to eventually migrate to sr.ht - but it's not ready yet.

In any case, I don't think we should rush into migrating right now. As @Ongy, I don't want to migrate to anything if it's to migrate again after that.

**Comment by Hyleus** [comment]
Could someone tell me why my comments get marked as spam? I agree that my first comment wasn't perfect and I've edited it.

**Comment by ascent12** [comment]
This is a discussion for the sway team, and has no bearing on end users. The opinions of those don't have some stake in sway/wlroots aren't helpful at all.

**Comment by SirCmpwn** [comment]
The plan is to move sway and wlroots to sr.ht - this has been the plan for longer than there have been talks of Microsoft buying GitHub. However, in light of the recent news, I intend to speed up work on sr.ht so that people leaving GitHub have another choice. To this end, I want to work on bringing sway+wlroots over sooner rather than later.

I know that it's not ready yet and that it would be disruptive to the project to move now. Can I get some feedback from core contributors on what you think sr.ht needs before it's suitable for sway+wlroots? If you don't already have an account please reach out and ask for one.

The biggest one is obviously public code review tools, ideally which don't require git-send-email to use.

**Comment by emersion** [comment]
Another proposal is to move to FreeDesktop.Org's GitLab instance, alongside with the rest of the FDO projects. A blocker for this is proper builds.sr.ht integration in GitLab (not possible right now because of GitLab MR limitations).

**Comment by emersion** [comment]
I've been working on a small project to overcome the GitLab API limitations: https://git.sr.ht/~emersion/dalligi

It allows exposing builds.sr.ht as a GitLab runner. This should bridge the gap.

It doesn't yet support multiple build manifests, nor does it mirror build logs. But that's planned.

**Comment by emersion** [comment]
https://gitlab.freedesktop.org/freedesktop/freedesktop/-/issues/473

**Comment by ddevault** [comment]
Dunno what this project is but maybe you guys should consider ~~SourceHut~~ Codeberg
